manifests/charts/cloudcore/templates/service_cloudcore.yaml

```diff
@@ -10,39 +10,35 @@ metadata:
   {{- end }}
   name: cloudcore
 spec:
-  {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
   type: {{ .Values.cloudCore.service.type }}
-  {{- else }}
-  type: ClusterIP
-  {{- end }}
   ports:
   - port: 10000
     targetPort: 10000
-    {{- if and (eq .Values.cloudCore.service.type "NodePort") ( not .Values.cloudCore.hostNetWork) }}
+    {{- if eq .Values.cloudCore.service.type "NodePort" }}
     nodePort: {{ .Values.cloudCore.service.cloudhubNodePort }}
     {{- end }}
```

Comment:
Oh, the hostNetWork mode will not use nodePort field. Am I right?

Reply:
cc  @lwabish

Reply:
Yes, you are right.But the point is, is it better to expose these fields directly to chart users?
For example,nodePort field is needed by LoadBalancer type service. 
As we can see now, the logics here in our chart can't cover this situation, which is also the motivation of this PR.
So in my opinion, it would be much more flexible to minimize our logic in charts unless necessary.

Reply:
have you tried hostNetWork: false to satisfy your senses？

Reply:
Just tried using LoadBalancer type service to expose cloudcore tcp and https ports to get an edgecore joined successfully.
According to the reply of issue #4862 ,hostnetwork and priviledged is not necessary for cloudcore. I dont't know why they are exposed, maybe for some special use cases.

Reply:
> Just tried using LoadBalancer type service to expose cloudcore tcp and https ports to get an edgecore joined successfully. According to the reply of issue #4862 ,hostnetwork and priviledged is not necessary for cloudcore. I dont't know why they are exposed, maybe for some special use cases.

CC @fisherxu maybe NodePort as the default choice is more suitable?

Reply:
When stream is enabled (logs/exec) and IptablesManager mode is 'internal', The Cloudcore hostNetwork should be true.

Reply:
Got it! I think the bottleneck is the current logic doesn't support Loadbalancer type service.